### PR TITLE
feat: Python VRP quickstart without time windows

### PR DESCRIPTION
**Pull request by Christopher-Chianelli**
### Development

- [ ] The changes have been covered with tests, if necessary.
- [ ] You have a green build, with the exception of the flaky tests.
- [ ] UI and JS files are fully tested, the user interface works for all modules affected by your changes (e.g., solve and analyze buttons).
- [ ] The network calls work for all modules affected by your changes (e.g., solving a problem).
- [ ] The console messages are validated for all modules affected by your changes.

### Code Review

- [ ] This pull request includes an explanatory title and description.
- [ ] The GitHub issue is linked.
- [ ] At least one other engineer has approved the changes.
- [ ] After PR is merged, inform the reporter.

### CODE REVIEW

File: python/vehicle-routing/README.adoc [hunk 1]
```diff
@@ -0,0 +1,79 @@
+= Employee Scheduling (Python)
```
Comment (zepfred) [inline python/vehicle-routing/README.adoc]: Let's fix the title.

File: python/vehicle-routing/src/vehicle_routing/constraints.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+##############################################
+
+
+def vehicle_capacity(factory: ConstraintFactory):
+    return (factory.for_each(Vehicle)
+            .filter(lambda vehicle: vehicle.calculate_total_demand() > vehicle.capacity)
+            .penalize(HardSoftScore.ONE_HARD,
+                      lambda vehicle: vehicle.calculate_total_demand() - vehicle.capacity)
+            .justify_with(lambda vehicle, score:
+                          VehicleCapacityJustification(
```
Comment (zepfred) [inline python/vehicle-routing/src/vehicle_routing/constraints.py:28]: It looks like the capacity param comes first.

File: python/vehicle-routing/static/recommended-fit.js [hunk 3]
```diff
@@ -0,0 +1,165 @@
+function addNewVisit(id, lat, lng, map, marker) {
```
Comment (zepfred) [inline python/vehicle-routing/static/recommended-fit.js]: Do we need this file?

**Comment by zepfred** [comment]
The Timefold logo is missing. Could you please double-check that?
Screenshot from 2024-06-17 14-15-13

**Comment by zepfred** [comment]
I'm getting the following when testing with Python 3.10:

```
(.venv) fred@fedora:~/Projects/timefold/timefold-quickstarts/python/vehicle-routing$ run-app
INFO:     @ uvicorn.error : Started server process [49004]
INFO:     @ uvicorn.error : Waiting for application startup.
INFO:     @ uvicorn.error : Application startup complete.
INFO:     @ uvicorn.error : Uvicorn running on http://127.0.0.1:8080 (Press CTRL+C to quit)
INFO:     @ uvicorn.access : 127.0.0.1:50206 - "GET / HTTP/1.1" 200
INFO:     @ uvicorn.access : 127.0.0.1:50206 - "GET /timefold/css/timefold-webui.css HTTP/1.1" 304
INFO:     @ uvicorn.access : 127.0.0.1:50206 - "GET /q/swagger-ui HTTP/1.1" 200
INFO:     @ uvicorn.access : 127.0.0.1:50206 - "GET /timefold/js/timefold-webui.js HTTP/1.1" 304
INFO:     @ uvicorn.access : 127.0.0.1:50216 - "GET /score-analysis.js HTTP/1.1" 304
INFO:     @ uvicorn.access : 127.0.0.1:50216 - "GET /app.js HTTP/1.1" 304
INFO:     @ uvicorn.access : 127.0.0.1:50216 - "GET /openapi.json HTTP/1.1" 200
INFO:     @ uvicorn.access : 127.0.0.1:50216 - "GET /demo-data HTTP/1.1" 200
INFO:     @ uvicorn.access : 127.0.0.1:50206 - "GET /webjars/timefold/img/timefold-logo-horizontal-negative.svg HTTP/1.1" 404
INFO:     @ uvicorn.access : 127.0.0.1:50206 - "GET /demo-data/PHILADELPHIA HTTP/1.1" 200
INFO:     @ uvicorn.access : 127.0.0.1:50206 - "POST /route-plans HTTP/1.1" 200
INFO:     @ timefold.solver : Solving started: time spent (35), best score (-55init/0hard/0soft), environment mode (REPRODUCIBLE), move thread count (NONE), random (JDK with seed 0).
INFO:     @ timefold.solver : Problem scale: entity count (6), variable count (6), approximate value count (55), approximate problem scale (6.934156 × 10^79).
INFO:     @ timefold.solver : Construction Heuristic phase (0) ended: time spent (320), best score (0hard/-14625soft), score calculation speed (6505/sec), step total (55).
INFO:     @ uvicorn.access : 127.0.0.1:50206 - "GET /route-plans/ID HTTP/1.1" 500
ERROR:    @ uvicorn.error : Exception in ASGI application
Traceback (most recent call last):
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/uvicorn/protocols/http/httptools_impl.py", line 399, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 70, in __call__
    return await self.app(scope, receive, send)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/fastapi/applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/applications.py", line 123, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/middleware/errors.py", line 186, in __call__
    raise exc
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/middleware/errors.py", line 164, in __call__
    await self.app(scope, receive, _send)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/middleware/exceptions.py", line 65, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
    raise exc
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    await app(scope, receive, sender)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/routing.py", line 756, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/routing.py", line 776, in app
    await route.handle(scope, receive, send)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/routing.py", line 297, in handle
    await self.app(scope, receive, send)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/routing.py", line 77, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
    raise exc
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    await app(scope, receive, sender)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/routing.py", line 72, in app
    response = await func(request)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/fastapi/routing.py", line 278, in app
    raw_response = await run_endpoint_function(
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/fastapi/routing.py", line 191, in run_endpoint_function
    return await dependant.call(**values)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/src/vehicle_routing/routes.py", line 52, in get_route
    solution_manager.update(out)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/timefold/solver/_solution_manager.py", line 64, in update
    java_solution = convert_to_java_python_like_object(solution)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/_jpyinterpreter/conversions.py", line 483, in convert_to_java_python_like_object
    out = convert_object_to_java_python_like_object(value, instance_map)
  File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/_jpyinterpreter/conversions.py", line 296, in convert_object_to_java_python_like_object
    CPythonBackedPythonInterpreter.updateJavaObjectFromPythonObject(out,
RuntimeError: dictionary changed size during iteration
```

The error seems similar to the one from `school-timetabling` module.

**Comment by Christopher-Chianelli** [comment]
> I'm getting the following when testing with Python 3.10:
> 
> ```
> (.venv) fred@fedora:~/Projects/timefold/timefold-quickstarts/python/vehicle-routing$ run-app
> INFO:     @ uvicorn.error : Started server process [49004]
> INFO:     @ uvicorn.error : Waiting for application startup.
> INFO:     @ uvicorn.error : Application startup complete.
> INFO:     @ uvicorn.error : Uvicorn running on http://127.0.0.1:8080 (Press CTRL+C to quit)
> INFO:     @ uvicorn.access : 127.0.0.1:50206 - "GET / HTTP/1.1" 200
> INFO:     @ uvicorn.access : 127.0.0.1:50206 - "GET /timefold/css/timefold-webui.css HTTP/1.1" 304
> INFO:     @ uvicorn.access : 127.0.0.1:50206 - "GET /q/swagger-ui HTTP/1.1" 200
> INFO:     @ uvicorn.access : 127.0.0.1:50206 - "GET /timefold/js/timefold-webui.js HTTP/1.1" 304
> INFO:     @ uvicorn.access : 127.0.0.1:50216 - "GET /score-analysis.js HTTP/1.1" 304
> INFO:     @ uvicorn.access : 127.0.0.1:50216 - "GET /app.js HTTP/1.1" 304
> INFO:     @ uvicorn.access : 127.0.0.1:50216 - "GET /openapi.json HTTP/1.1" 200
> INFO:     @ uvicorn.access : 127.0.0.1:50216 - "GET /demo-data HTTP/1.1" 200
> INFO:     @ uvicorn.access : 127.0.0.1:50206 - "GET /webjars/timefold/img/timefold-logo-horizontal-negative.svg HTTP/1.1" 404
> INFO:     @ uvicorn.access : 127.0.0.1:50206 - "GET /demo-data/PHILADELPHIA HTTP/1.1" 200
> INFO:     @ uvicorn.access : 127.0.0.1:50206 - "POST /route-plans HTTP/1.1" 200
> INFO:     @ timefold.solver : Solving started: time spent (35), best score (-55init/0hard/0soft), environment mode (REPRODUCIBLE), move thread count (NONE), random (JDK with seed 0).
> INFO:     @ timefold.solver : Problem scale: entity count (6), variable count (6), approximate value count (55), approximate problem scale (6.934156 × 10^79).
> INFO:     @ timefold.solver : Construction Heuristic phase (0) ended: time spent (320), best score (0hard/-14625soft), score calculation speed (6505/sec), step total (55).
> INFO:     @ uvicorn.access : 127.0.0.1:50206 - "GET /route-plans/ID HTTP/1.1" 500
> ERROR:    @ uvicorn.error : Exception in ASGI application
> Traceback (most recent call last):
>   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/uvicorn/protocols/http/httptools_impl.py", line 399, in run_asgi
>     result = await app(  # type: ignore[func-returns-value]
>   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 70, in __call__
>     return await self.app(scope, receive, send)
>   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/fastapi/applications.py", line 1054, in __call__
>     await super().__call__(scope, receive, send)
>   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/applications.py", line 123, in __call__
>     await self.middleware_stack(scope, receive, send)
>   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/middleware/errors.py", line 186, in __call__
>     raise exc
>   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/middleware/errors.py", line 164, in __call__
>     await self.app(scope, receive, _send)
>   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/middleware/exceptions.py", line 65, in __call__
>     await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
>   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
>     raise exc
>   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
>     await app(scope, receive, sender)
>   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/routing.py", line 756, in __call__
>     await self.middleware_stack(scope, receive, send)
>   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/routing.py", line 776, in app
>     await route.handle(scope, receive, send)
>   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/routing.py", line 297, in handle
>     await self.app(scope, receive, send)
>   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/routing.py", line 77, in app
>     await wrap_app_handling_exceptions(app, request)(scope, receive, send)
>   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
>     raise exc
>   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
>     await app(scope, receive, sender)
>   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/routing.py", line 72, in app
>     response = await func(request)
>   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/fastapi/routing.py", line 278, in app
>     raw_response = await run_endpoint_function(
>   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/fastapi/routing.py", line 191, in run_endpoint_function
>     return await dependant.call(**values)
>   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/src/vehicle_routing/routes.py", line 52, in get_route
>     solution_manager.update(out)
>   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/timefold/solver/_solution_manager.py", line 64, in update
>     java_solution = convert_to_java_python_like_object(solution)
>   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/_jpyinterpreter/conversions.py", line 483, in convert_to_java_python_like_object
>     out = convert_object_to_java_python_like_object(value, instance_map)
>   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/_jpyinterpreter/conversions.py", line 296, in convert_object_to_java_python_like_object
>     CPythonBackedPythonInterpreter.updateJavaObjectFromPythonObject(out,
> RuntimeError: dictionary changed size during iteration
> ```
> 
> The error seems similar to the one from `school-timetabling` module.

Just tested 3.10; cannot reproduce locally

**Comment by zepfred** [comment]
> > I'm getting the following when testing with Python 3.10:
> > ```
> > (.venv) fred@fedora:~/Projects/timefold/timefold-quickstarts/python/vehicle-routing$ run-app
> > INFO:     @ uvicorn.error : Started server process [49004]
> > INFO:     @ uvicorn.error : Waiting for application startup.
> > INFO:     @ uvicorn.error : Application startup complete.
> > INFO:     @ uvicorn.error : Uvicorn running on http://127.0.0.1:8080 (Press CTRL+C to quit)
> > INFO:     @ uvicorn.access : 127.0.0.1:50206 - "GET / HTTP/1.1" 200
> > INFO:     @ uvicorn.access : 127.0.0.1:50206 - "GET /timefold/css/timefold-webui.css HTTP/1.1" 304
> > INFO:     @ uvicorn.access : 127.0.0.1:50206 - "GET /q/swagger-ui HTTP/1.1" 200
> > INFO:     @ uvicorn.access : 127.0.0.1:50206 - "GET /timefold/js/timefold-webui.js HTTP/1.1" 304
> > INFO:     @ uvicorn.access : 127.0.0.1:50216 - "GET /score-analysis.js HTTP/1.1" 304
> > INFO:     @ uvicorn.access : 127.0.0.1:50216 - "GET /app.js HTTP/1.1" 304
> > INFO:     @ uvicorn.access : 127.0.0.1:50216 - "GET /openapi.json HTTP/1.1" 200
> > INFO:     @ uvicorn.access : 127.0.0.1:50216 - "GET /demo-data HTTP/1.1" 200
> > INFO:     @ uvicorn.access : 127.0.0.1:50206 - "GET /webjars/timefold/img/timefold-logo-horizontal-negative.svg HTTP/1.1" 404
> > INFO:     @ uvicorn.access : 127.0.0.1:50206 - "GET /demo-data/PHILADELPHIA HTTP/1.1" 200
> > INFO:     @ uvicorn.access : 127.0.0.1:50206 - "POST /route-plans HTTP/1.1" 200
> > INFO:     @ timefold.solver : Solving started: time spent (35), best score (-55init/0hard/0soft), environment mode (REPRODUCIBLE), move thread count (NONE), random (JDK with seed 0).
> > INFO:     @ timefold.solver : Problem scale: entity count (6), variable count (6), approximate value count (55), approximate problem scale (6.934156 × 10^79).
> > INFO:     @ timefold.solver : Construction Heuristic phase (0) ended: time spent (320), best score (0hard/-14625soft), score calculation speed (6505/sec), step total (55).
> > INFO:     @ uvicorn.access : 127.0.0.1:50206 - "GET /route-plans/ID HTTP/1.1" 500
> > ERROR:    @ uvicorn.error : Exception in ASGI application
> > Traceback (most recent call last):
> >   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/uvicorn/protocols/http/httptools_impl.py", line 399, in run_asgi
> >     result = await app(  # type: ignore[func-returns-value]
> >   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 70, in __call__
> >     return await self.app(scope, receive, send)
> >   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/fastapi/applications.py", line 1054, in __call__
> >     await super().__call__(scope, receive, send)
> >   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/applications.py", line 123, in __call__
> >     await self.middleware_stack(scope, receive, send)
> >   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/middleware/errors.py", line 186, in __call__
> >     raise exc
> >   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/middleware/errors.py", line 164, in __call__
> >     await self.app(scope, receive, _send)
> >   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/middleware/exceptions.py", line 65, in __call__
> >     await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
> >   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
> >     raise exc
> >   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
> >     await app(scope, receive, sender)
> >   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/routing.py", line 756, in __call__
> >     await self.middleware_stack(scope, receive, send)
> >   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/routing.py", line 776, in app
> >     await route.handle(scope, receive, send)
> >   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/routing.py", line 297, in handle
> >     await self.app(scope, receive, send)
> >   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/routing.py", line 77, in app
> >     await wrap_app_handling_exceptions(app, request)(scope, receive, send)
> >   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
> >     raise exc
> >   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
> >     await app(scope, receive, sender)
> >   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/starlette/routing.py", line 72, in app
> >     response = await func(request)
> >   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/fastapi/routing.py", line 278, in app
> >     raw_response = await run_endpoint_function(
> >   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/fastapi/routing.py", line 191, in run_endpoint_function
> >     return await dependant.call(**values)
> >   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/src/vehicle_routing/routes.py", line 52, in get_route
> >     solution_manager.update(out)
> >   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/timefold/solver/_solution_manager.py", line 64, in update
> >     java_solution = convert_to_java_python_like_object(solution)
> >   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/_jpyinterpreter/conversions.py", line 483, in convert_to_java_python_like_object
> >     out = convert_object_to_java_python_like_object(value, instance_map)
> >   File "/home/fred/Projects/timefold/timefold-quickstarts/python/vehicle-routing/.venv/lib64/python3.10/site-packages/_jpyinterpreter/conversions.py", line 296, in convert_object_to_java_python_like_object
> >     CPythonBackedPythonInterpreter.updateJavaObjectFromPythonObject(out,
> > RuntimeError: dictionary changed size during iteration
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The error seems similar to the one from `school-timetabling` module.
> 
> Just tested 3.10; cannot reproduce locally

@triceo, can you try to reproduce it?

**Comment by triceo** [comment]
@zepfred I can look into it tomorrow afternoon, which may be too late.

**Comment by zepfred** [comment]
One final comment regarding the test dataset. The final solution shows that one of the vehicles has no scheduled visits.

Screenshot from 2024-06-17 14-25-34

Given our deadline for the next release, I'm fine with the current behavior. However, I believe there is room for improvement to make it more realistic and ensure that all vehicles are used. Here is the Java result:

image

**Comment by Christopher-Chianelli** [comment]
> One final comment regarding the test dataset. The final solution shows that one of the vehicles has no scheduled visits.
> 
> Screenshot from 2024-06-17 14-25-34
> 
> Given our deadline for the next release, I'm fine with the current behavior. However, I believe there is room for improvement to make it more realistic and ensure that all vehicles are used. Here is the Java result:
> 
> image

That might be because using that vehicle increase distance, and it is not needed for capacity (this VRP does not take into account time windows)

**Comment by Christopher-Chianelli** [comment]
(In particular, there is no penalty/reward for using all vehicles; if anything, it a suggestion that one vehicle might not be needed to serve all requests)

**Comment by zepfred** [comment]
> > One final comment regarding the test dataset. The final solution shows that one of the vehicles has no scheduled visits.
> > Screenshot from 2024-06-17 14-25-34
> > Given our deadline for the next release, I'm fine with the current behavior. However, I believe there is room for improvement to make it more realistic and ensure that all vehicles are used. Here is the Java result:
> > image
> 
> That might be because using that vehicle increase distance, and it is not needed for capacity (this VRP does not take into account time windows)

I understand. So, it means that it will eventually be solved once we add support for time windows, right?

**Comment by Christopher-Chianelli** [comment]
> > > One final comment regarding the test dataset. The final solution shows that one of the vehicles has no scheduled visits.
> > > Screenshot from 2024-06-17 14-25-34
> > > Given our deadline for the next release, I'm fine with the current behavior. However, I believe there is room for improvement to make it more realistic and ensure that all vehicles are used. Here is the Java result:
> > > image
> > 
> > 
> > That might be because using that vehicle increase distance, and it is not needed for capacity (this VRP does not take into account time windows)
> 
> I understand. So, it means that it will eventually be solved once we add support for time windows, right?

Since time windows make the problem harder, probably.

**Comment by zepfred** [comment]
The root README file does not reference the Python modules, but it does for Kotlin. Should we do that for Python?

image
image

**Comment by triceo** [comment]
@zepfred Good point about the README.

The structure of the README no longer matches the structure of the repo.
Before, we had tech and use case. Now the categories are Java / Kotlin / Python.

We should change the table.

**Comment by triceo** [comment]
That said, for the July release, this should be refactored much more; the README is becoming impossibly long.

**Review by zepfred — APPROVED** [review]
The logic looks correct, but I can't run it with Python 3.10, which may be an error on my end. @triceo will attempt to replicate it, so we can have another validation.

Another important point is the need to refactor the root README file, including references to Java, Kotlin, and Python. However, this work should be addressed by a different PR.

**Comment by zepfred** [comment]
@triceo, the issue is related to the call `solution_manager.update`, which will require further investigation. @Christopher-Chianelli  has removed the call, so there is no need to reproduce the issue at the moment.

**Comment by Christopher-Chianelli** [comment]
> @triceo, the issue is related to the call `solution_manager.update`, which will require further investigation. @Christopher-Chianelli has removed the call, so there is no need to reproduce the issue at the moment.

Additionally, said call was pointless and was an artifact of the previous version of the quickstart

**Pull request merged**